base/math.jl

```diff
@@ -845,6 +845,19 @@ function _hypot(x::NTuple{N,<:Number}) where {N}
     end
 end
 
+function _hypot(x::NTuple{N,T}) where {N,T<:IEEEFloat}
+    infT = convert(T, Inf)
+    x = abs.(x) # doesn't change result but enables computational shortcuts
+    # note: any() was causing this to not inline for N=3 but mapreduce() was not
+    mapreduce(==(infT), |, x) && return infT # return Inf even if an argument is NaN
+    maxabs = reinterpret(T, maximum(z -> reinterpret(Signed, z), x)) # for abs(::IEEEFloat) values, a ::BitInteger cast does not change the result
+    maxabs > zero(T) || return maxabs # catch NaN before the @fastmath below, but also shortcut 0 since we can (remove if no more @fastmath below)
+    scale,invscale = scaleinv(maxabs)
+     # @fastmath(+) to allow reassociation (see #48129)
+    add_fast(x, y) = Core.Intrinsics.add_float_fast(x, y) # @fastmath is not available during bootstrap
+    return scale * sqrt(mapreduce(y -> abs2(y * invscale), add_fast, x))
```

Comment:
Can't you use `Core.Intrinsics.add_float_fast` directly?

Reply:
No. This results in dynamic dispatch for reasons I haven't explored.
```julia-repl
  123.996 ns (10 allocations: 160 bytes)
  165.282 ns (13 allocations: 208 bytes)
  327.826 ns (25 allocations: 400 bytes)
  674.839 ns (49 allocations: 784 bytes)
```

Reply:
hmm. can you open an issue for this?

Reply:
Not supprising as
```julia
julia> typeof(Core.Compiler.add_float_fast)
Core.IntrinsicFunction
```